my_game.py

```diff
@@ -45,6 +45,54 @@ def wrap(sprite: arcade.Sprite):
         sprite.center_y -= CONFIG['SCREEN_HEIGHT']
 
 
+class Shot(arcade.Sprite):
+    """
+    universal class for shot objects
+    """
+
+    def __init__(self, filename, center_x, center_y, angle, speed, range, sound=None):
+
+        super().__init__(
+            filename=filename,
+            scale=CONFIG['SPRITE_SCALING'],
+            center_x=center_x,
+            center_y=center_y,
+            angle=angle,
+            flipped_horizontally=True,
+            flipped_diagonally=True
+            )
+
+        self.speed = speed
+        self.range = range
+        self.distance_traveled = 0
+
+        self.forward(self.speed)
+
+        # play the shot sound if present
+        if sound:
+            sound.play()
+
+    def update(self):
```

Comment:
We should use _on_update()_ with _delta_time_

Reply:
well, it isnt so in our other obj-classes... except for player - so we should shift it everywhere?

Reply:
Yes, we should use _on_update()_ everywhere.